sympy/stats/crv.py

```diff
@@ -335,13 +335,25 @@ def where(self, condition):
         interval = interval.intersect(self.domain.set)
         return SingleContinuousDomain(rv.symbol, interval)
 
+
     def conditional_space(self, condition, normalize=True, **kwargs):
 
         condition = condition.xreplace(dict((rv, rv.symbol) for rv in self.values))
+        domain    = ConditionalContinuousDomain(self.domain, condition)
 
-        domain = ConditionalContinuousDomain(self.domain, condition)
         if normalize:
-            pdf = self.pdf / domain.integrate(self.pdf, **kwargs)
+            # create a clone of the variable to
+            # make sure that variables in nested integrals are different.
+            # this makes sure that they are evaluated separately
+            to_dummy    = lambda rv: Dummy(str(rv.symbol))
+            replacement = {rv.symbol: to_dummy(rv) for rv in self.values}
+            clone       = self.xreplace(replacement)
+            clone_domain = ConditionalContinuousDomain(
+                                clone.domain, condition.xreplace(replacement))
+
+            normalizer = clone_domain.integrate(clone.pdf, **kwargs)
+
+            pdf = self.pdf / normalizer
```

Comment:
It would probably be more efficient to compute `normalizer` here without cloning. Is there some reason why that should not be done?

Reply:
Without cloning the unevaluated integral looks like this.
![undifferentiated](https://user-images.githubusercontent.com/15707729/30814150-64e94888-a1dd-11e7-966a-78cdedb2222e.png)
This, of course, evaluates to S.NaN With cloning it looks like this and evaluates to the correct value.
![differentiated](https://user-images.githubusercontent.com/15707729/30814147-630bf8ee-a1dd-11e7-8162-caca82e0be38.png)
The problem is that without cloning integrate() thinks that the mu in the inside integral and mu in the outside the integral are the same mu. This made it perform some bound optimizations that result in the pdf being integrated over 0/0 = S.NaN outside of the domain of the random variable. The trick is to replace each variable with a dummy.

I'm fixing other style issues and will make another commit once the tests are done running.

Reply:
It should be possible to evaluate the normalizer by `.doit()` even without cloning. Once the integral has been evaluated (to 1/2 in this case) there would be no integration variable left in the denominator and no confusion would occur.

Reply:
You're right. If the user wants the result evaluated, then it seems that this is what should be done. However, if the user wants the symbolic output, then the random variable needs to be cloned. I want to keep everything as symbolic as possible and, as much as possible, leave the user with the choice to defer evaluation.